### PR TITLE
update admiral/api/services/wwwConfig.js to set SHIPPABLE_PUBLIC_API_URL and SHIPPABLE_INT_API_URL

### PR DESCRIPTION
**Pull request by ankul-shippable**
#2268

```
 "Env": [
                "SHIPPABLE_API_URL=http://172.17.0.1:50000",
                "SHIPPABLE_PUBLIC_API_URL=http://172.17.0.1:50000",
                "SHIPPABLE_INT_API_URL=http://172.17.0.1:50000",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "BASH_ENV=/etc/drydock/.env"
            ],

```

### CODE REVIEW

File: api/services/wwwConfig.js [hunk 1]
```diff
@@ -162,7 +165,11 @@ function _generateEnvs(bag, next) {
   envs = util.format('%s -e %s=%s',
     envs, 'SHIPPABLE_API_TOKEN', bag.serviceUserToken);
   envs = util.format('%s -e %s=%s',
-    envs, 'SHIPPABLE_API_URL', apiUrl);
+    envs, 'SHIPPABLE_API_URL', publicApiUrl);
```
Comment (ankul-shippable) [inline api/services/wwwConfig.js]: will remove this once www is updated to use new ENVs

File: api/services/wwwConfig.js [hunk 2]
```diff
@@ -123,12 +123,10 @@ function _getAPISystemIntegration(bag, next) {
             'No systemIntegration found for query: ' + query)
         );
 
-      var defaultAPIIntegration = _.findWhere(
+      bag.publicApiIntegration = _.findWhere(
         systemIntegrations, {name: 'api'});
-      var serviceAPIIntegration = _.findWhere(
```
Comment (shrivara) [inline api/services/wwwConfig.js]: Do we want to remove this? I'm not quite sure what the original purpose of this was.

File: api/services/wwwConfig.js [hunk 3]
```diff
@@ -149,10 +147,15 @@ function _generateEnvs(bag, next) {
   var who = bag.who + '|' + _generateEnvs.name;
   logger.verbose(who, 'Inside');
 
-  var apiUrl = bag.apiSystemIntegration.data &&
-    bag.apiSystemIntegration.data.url;
+  var publicApiUrl = bag.publicApiIntegration.data &&
+    bag.publicApiIntegration.data.url;
+
+  var internalApiUrl = bag.internalApiIntegration.data &&
```
Comment (shrivara) [inline api/services/wwwConfig.js]: Is `bag.internalApiIntegration` always defined?

File: api/services/wwwConfig.js [hunk 4]
```diff
@@ -123,12 +123,10 @@ function _getAPISystemIntegration(bag, next) {
             'No systemIntegration found for query: ' + query)
         );
 
-      var defaultAPIIntegration = _.findWhere(
+      bag.publicApiIntegration = _.findWhere(
         systemIntegrations, {name: 'api'});
-      var serviceAPIIntegration = _.findWhere(
-        systemIntegrations, {name: bag.config.apiUrlIntegration});
-
-      bag.apiSystemIntegration = serviceAPIIntegration || defaultAPIIntegration;
```
Comment (Bharath92) [inline api/services/wwwConfig.js]: i guess u can keep this. `bag.internalApiIntegration = bag.internalApiIntegration || bag.publicApiIntegration;`

File: api/services/wwwConfig.js (hunk 1 above)
Comment (Bharath92) [reply api/services/wwwConfig.js]: why remove this? `SHIPPABLE_API_URL` is public api url right? Is the reason, to make public api url env more explicit?

File: api/services/wwwConfig.js (hunk 2 above)
Comment (ankul-shippable) [reply api/services/wwwConfig.js]: I think this was there to make sure www uses the api Url from the systemIntegration defined in config.json

File: api/services/wwwConfig.js (hunk 4 above)
Comment (ankul-shippable) [reply api/services/wwwConfig.js]: ok, I'll update

File: api/services/wwwConfig.js (hunk 3 above)
Comment (ankul-shippable) [reply api/services/wwwConfig.js]: yeah, it will always be defined

File: api/services/wwwConfig.js (hunk 1 above)
Comment (ankul-shippable) [reply api/services/wwwConfig.js]: yes, we want to make it more explicit to the users.

File: api/services/wwwConfig.js [hunk 5]
```diff
@@ -149,10 +149,15 @@ function _generateEnvs(bag, next) {
...
 
-  var apiUrl = bag.apiSystemIntegration.data &&
-    bag.apiSystemIntegration.data.url;
+  var publicApiUrl = bag.publicApiIntegration.data &&
+    bag.publicApiIntegration.data.url;
+
+  var internalApiUrl = bag.internalApiIntegration.data &&
+    bag.internalApiIntegration.data.url;
+
+  internalApiUrl = internalApiUrl || publicApiUrl;
```
Comment (shrivara) [inline api/services/wwwConfig.js]: This is probably not necessary any more.

Comment (ankul-shippable) [reply api/services/wwwConfig.js]: updated

**Pull request merged**